File: aoc/2023/ocaml/bin/main.ml
(*
   dune exec --no-print-directory ./main.exe < ../../_inputs/day01.txt
*)

module Curr = Aoc.Day01
open Printf

let () =
  let input = In_channel.input_all In_channel.stdin in
  let lines = String.split_on_char '\n' input in
  ()
  ; printf "Part 1: %d\n" @@ Curr.part1 lines
  ; printf "Part 2: %d\n" @@ Curr.part2 lines
;;

File: aoc/2023/rust/src/main.rs
mod day_01;

/*

rg --files -t rust | entr -c cargo check
cargo run < ../_inputs/day01.txt
rg --files -t rust | entr -c bash -c 'cargo run < ../_inputs/day01.txt'

*/

fn main() {
    // Not sure how to fix that
    let lines1: Vec<String> = std::io::stdin().lines().map(|line| line.unwrap()).collect();
    let lines2: Vec<&str> = lines1.iter().map(|line| line.as_str()).collect();

    println!("Part 1: {}", day_01::part_1(lines2.clone()));
    println!("Part 2: {}", day_01::part_2(lines2.clone()));
}
